internal/pkg/template/diff/override.go

```diff
@@ -40,58 +47,173 @@ func (m *ignorer) parse(_, _ *yaml.Node, _ string, _ overrider) (diffNode, error
+// input node to "GetAtt".
+// Example1: "!GetAtt" and "!GetAtt" returns true.
+// Example2: "!GetAtt" and "Fn::GetAtt" returns true.
+// Example3: "!Ref" and "!GetAtt" returns false.
+// Example4: "!GetAtt [a,b]" and "Fn::GetAtt: a:b" returns false because the input type is wrong.
+func (converter *getAttConverter) match(from, to *yaml.Node, key string, overrider overrider) bool {
+	if !converter.intrinsicFunc.match(from, to, key, overrider) {
+		return false
+	}
+	if intrinsicFuncName(from) != "GetAtt" {
+		return false
+	}
+	fromValue, toValue := from, to
+	if from.Kind == yaml.MappingNode {
+		// A valid full-form intrinsic function always contain a child node.
+		// This must be valid because it has passed `converter.intrinsicFunc.match`.
+		fromValue = from.Content[1]
+	}
+	if to.Kind == yaml.MappingNode {
+		toValue = to.Content[1]
+	}
+	return (fromValue.Kind == yaml.ScalarNode || fromValue.Kind == yaml.SequenceNode) && (toValue.Kind == yaml.ScalarNode || toValue.Kind == yaml.SequenceNode)
+}
+
+// parse compares two nodes that call the "GetAtt" function. Both from and to can be written in either full or short form.
+// parse assumes that from and to are already matched by getAttConverter.
+func (converter *getAttConverter) parse(from, to *yaml.Node, key string, overrider overrider) (diffNode, error) {
+	// Extract the input node to GetAtt.
+	fromValue, toValue := from, to
+	if from.Kind == yaml.MappingNode {
+		fromValue = from.Content[1] // A valid full-form intrinsic function always contain a child node.
+	}
+	if to.Kind == yaml.MappingNode {
+		toValue = to.Content[1]
+	}
+	// If the input node are of the same type (i.e. both seq or both scalar), parse them normally.
+	if fromValue.Kind == toValue.Kind {
+		return converter.intrinsicFuncMapTagConverter.parse(from, to, key, overrider)
+	}
+	// Otherwise, first convert the scalar input to seq input, then parse.
+	var err error
+	switch {
+	case fromValue.Kind == yaml.ScalarNode:
+		fromValue, err = getAttScalarToSeq(fromValue)
+	case toValue.Kind == yaml.ScalarNode:
+		toValue, err = getAttScalarToSeq(toValue)
 	}
 	if err != nil {
 		return nil, err
 	}
+	diff, err := parse(fromValue, toValue, intrinsicFunctions["GetAtt"].fullName, overrider)
 	if diff == nil {
-		return nil, nil
+		return nil, err
 	}
 	return &keyNode{
 		keyValue:   key,
 		childNodes: []diffNode{diff},
 	}, nil
 }
 
+func intrinsicFuncName(node *yaml.Node) string {
+	if node.Kind != yaml.MappingNode {
+		return strings.TrimPrefix(intrinsicFuncShortFormRegex.FindString(node.Tag), "!")
+	}
+	if len(node.Content) != 2 {
+		// The full form mapping node always contain only one child node, whose key is the func name in full form.
+		// Read https://www.efekarakus.com/2020/05/30/deep-dive-go-yaml-cfn.html.
+		return ""
+	}
+	return strings.TrimPrefix(intrinsicFuncFullFormRegex.FindString(node.Content[0].Value), "Fn::")
```

Comment:
i think this would end up returning the same thing for these two values of `node.Content[0].Value`: `Fn::Foo` and `Fn::Foo-bar`. i'm probably just being pedantic, but i wonder if we should use the regex to see if the value _matches_ our expected form, and then just trim `Fn::` off of that?

Reply:
> returning the same thing 

There is a `$` anchor at the end of regex though, so `Fn::Foo-bar` would give `Fn::Foo-bar` (it can't partially match `"Fn::Foo"` because of the `$`). Does this solve your concern or could you elaborate 👁️ ?

Reply:
```
reg := regexp.MustCompile(`^Fn::[A-Z][A-Za-z\d]*|Ref$`)
fmt.Printf("str: %q\n", reg.FindString("Fn::Foo-bar"))
fmt.Printf("str: %q\n", reg.FindString("Fn::Foo"))
```

tbh i don't _super_ know how it works, but both of this prints:
```
str: "Fn::Foo"
str: "Fn::Foo"
```
but you're right, you have the `$`, so you would think that `Fn::Foo-bar` should return `""`...

Reply:
I see lol perhaps it's parsing as "either start with Fn::xxx"  or "Ref and end it"

Reply:
i think it's something weird with the `|` and different capture groups. I tried:
```
reg := regexp.MustCompile(`^((Fn::[A-Z][A-Za-z\d]*)|(Ref))$`)
fmt.Printf("str: %q\n", reg.FindString("Fn::Foo-bar"))
fmt.Printf("str: %q\n", reg.FindString("Fn::Foo"))
fmt.Printf("match: %v\n", reg.MatchString("Fn::Foo-bar"))
fmt.Printf("match: %v\n", reg.MatchString("Fn::Foo"))
```

output:
```
str: ""
str: "Fn::Foo"
match: false
match: true
```

Reply:
yeah I think it wouldn't match `-bar` because there is a dash there, and dash isn't matched by `[A-Za-z\d]`

Reply:
`Fn::Foo-bar` shouldn't match! like you said, `-` is not matched by `[A-Za-z\d]`, and the regex requires `[A-Za-z\d]` until the _end of the line_.